### DISCUSSION TITLE
Proposal submission ideas for `GSoC`

### DISCUSSION THREAD

**Discussion by didogrigorov** [wagtail/gsoc#90] [Ideas]
Hello guys at @wagtail and thanks for the opportunity to contribute to your product. I'd like to participate as a GSoC open-source contributor to Wagtail. 

I'm a Junior Python Developer with about 1 year of experience. I've just graduated from a software academy, entirely based on Python, Django and Javascript and I'm a big fan of Django. What I've been currently working on is an algorithm visualizer with Django and chart.js. You can test some visualizations here, please input a number in the range of 1-100: 

[https://www.algolizer.net/insertion-sort](https://www.algolizer.net/insertion-sort)
[https://www.algolizer.net/bubble](https://www.algolizer.net/bubble)
[https://www.algolizer.net/selection-sort](https://www.algolizer.net/selection-sort)

**I have some ideas for features to be added to Wagtail, but before submitting my proposal, I'd like to hear what you think about them:**

1. Comment system with an option to be turned on or off page by page or page type by page type.
    
2. Better basic SEO control options:
    - Indexing options right from the Wagtail panel page by page
    - Choosing which types of content to be indexed or not (by page type).
    - Creating meta titles and descriptions templates by page types to be spread out within similar pages.
    - Extending of the Schema.org markup according to the approved by Google types here: https://developers.google.com/search/docs/appearance/structured-data/search-gallery The should be able to choose the type of Schema markup according to the format of content they want to publish.
    - Direct visibility of the robots.txt file for editing by the users.
    - Sitemap regeneration

3. More extensive video embedding with the option for a full YouTube channel to be integrated

4. Creation of a basic e-commerce option for users who want to have basic shopping cart features.

5. Basic ChatGPT integration in Wagtail for faster writing of content by user criteria - might be also an API one where once the content is created to be transferred to Wagtail.

6. Option for adding custom HTML by the user in the rich text editor tool.

7. Option for adding tables to the rich editor features or to the blocks. (This might save lots of time especially for users not so proficient with HTML and CSS styling).

Let me know what you think.

_With regards,_

**Dido Grigorov**

**Comment by thibaudcolas** [comment]
Hi Dido! That’s a lot of ideas. How much time have you spent researching what’s possible in Wagtail? Quite a few of those exist already. I think some of those ideas are good but I don’t see them being successful GSoC project material at this point – they’re too small for the bits of this that aren’t already existing.

**Reply by didogrigorov** [reply]
Thanks for your prompt response! Actually I'm a beginner with Wagtail and have started currently building my own personal website based on it. I really love everything Django related, so if you don't like my ideas or you think they don't fit in, I'd love to participate in something you have on your mind or already planned. Just let me know and share more details.

I'm totally new to open-source contribution at all, but I'd love to be helpful and I'm very motivated to gain more knowledge along the process of contribution. I am particularly interested in participating in GSoC with Wagtail because I believe it will provide me with an opportunity to hone my skills, gain valuable experience, and make a significant impact on a real-world project.

With regards,

**Dido**

**Reply by Nathaniel330** [reply]
We're on the same page, sir. But unlike you I do not have any proposal idea at all! 

Just a little more, I'm gonna be asking the organisation, "What can I do for the team?" "Is there something you need help with?" 😅

**Reply by thibaudcolas** [reply]
@didogrigorov we have our list of pre-vetted [project ideas](https://github.com/wagtail/gsoc/blob/main/project-ideas.md) featured prominently in our GSoC profile. This is what I’d recommend focusing on, unless you have a specific idea you think is a great fit due to your own research.

From your list – if you wanted to propose one, I’d say the idea that has the most promise is AI integration with Wagtail, since that’s a very rapidly evolving area.
